chronicles.nim

```diff
@@ -255,6 +262,13 @@ macro logIMPL(lineInfo: static InstInfo,
   code.add newCall("logAllDynamicProperties", Stream, record)
   code.add newCall("flushRecord", record)
 
+  code = quote do:
```

Comment:
I would prefer if this is controlled by a compile-time option. Also, we can use one lock per stream I think (or should it be even per-sink/per-output?).

Reply:
> I would prefer if this is controlled by a compile-time option.

Done. All locking is disabled if threading is off.

> Also, we can use one lock per stream I think (or should it be even per-sink/per-output?).

The finer-grained your locking, the more runtime overhead you have, with no actual benefit. Then you have to worry about different threads being able to interlace their setProperty() calls (or any other sub-record component).

This way, with one lock per log call, you're sure your record will reach the output in one piece and the wait should be short enough to not matter.

Worth revisiting with some actual instrumentation data on a system with 32+ cores and see what happens when they all wait on the same lock to do their logging. At that point, we're going to ask Jacek about logging buffers and dedicated logging threads.